pyro/infer/mcmc/nuts.py

```diff
@@ -209,12 +225,16 @@ def _build_tree(self, z, r, z_grads, log_slice, direction, tree_depth, energy_cu
         # For the special case that the sizes of each half are both 0,
         #     we choose the proposal from the first half
         #     (any is fine, because the probability of picking it at the end is 0!).
-        if tree_size != 0:
```

Comment:
We can remove the comment above?

Reply:
It seems that we don't have to check these conditions. It will never happen because we return first half_tree when it is turning or diverging.

Reply:
Great, let us remove the comment in that case!

Reply:
Ah, when using slice_sampling, if sliced_energy > 0, we also ignore the point (though it is not both turning and diverging). So there are cases tree_size = 0.